cmd/loadTests.go

```diff
@@ -358,27 +289,46 @@ func setup(cmd *cobra.Command, args []string) {
 		metricsInstance.PrintResults()
 	}
 
-	err = createLogDataJSON(
-		"load-tests.json",
-		loadTestsTimestamp,
-		overallCount,
-		threadCount,
-		numberOfUsers,
-		userBatches,
-		"Completed",
-		averageTimeToSpinUpUsers,
-		averageTimeToCreateResources,
-		averageTimeToRunPipelines,
-		userCreationFailureCount,
-		userCreationFailurePercentage,
-		resourceCreationFailureCount,
-		resourceCreationFailurePercentage,
-		pipelineRunFailureCount,
-		PipelineRunFailurePercentage,
-	)
+	/*
+		fetch the below fields:
+		machineName string - the machine on-which the loadTests are run,
+		binaryDetails string - binary details of the program that runs the tests
+	*/
+	machineName, err := os.Hostname()
+	if err != nil {
+		fmt.Printf("error getting hostname: %v\n", err)
+		return
+	}
+
+	goVersion := runtime.Version()
+	goOS := runtime.GOOS
+	goArch := runtime.GOARCH
+	binaryDetails := fmt.Sprintf("Built with %s for %s/%s", goVersion, goOS, goArch)
+
+	logDataInput := LogData{
+		Timestamp:                         loadTestsTimestamp,
+		EndTimestamp:                      loadTestsEndTimestamp,
+		MachineName:                       machineName,
+		BinaryDetails:                     binaryDetails,
+		NumberOfThreads:                   threadCount,
+		NumberOfUsersPerThread:            numberOfUsers,
+		NumberOfUsers:                     overallCount,
+		BatchSize:                         userBatches,
+		LoadTestCompletionStatus:          "Completed",
+		AverageTimeToSpinUpUsers:          averageTimeToSpinUpUsers,
+		AverageTimeToCreateResources:      averageTimeToCreateResources,
+		AverageTimeToRunPipelines:         averageTimeToRunPipelines,
+		UserCreationFailureCount:          userCreationFailureCount,
+		UserCreationFailurePercentage:     userCreationFailurePercentage,
+		ResourceCreationFailureCount:      resourceCreationFailureCount,
+		ResourceCreationFailurePercentage: resourceCreationFailurePercentage,
+		PipelineRunFailureCount:           pipelineRunFailureCount,
+		PipelineRunFailurePercentage:      PipelineRunFailurePercentage,
+	}
 
+	err = createLogDataJSON("load-tests.json", logDataInput)
 	if err != nil {
-		fmt.Printf("error marshalling JSON: %v\n", err)
+		fmt.Printf("error while marshalling JSON: %v\n", err)
```

Comment:
@naftalysh Please use `klog.Errorf` to print errors instead as in the rest of the code. Thx
```suggestion
		klog.Errorf("error while marshalling JSON: %v\n", err)
```